### PR TITLE
Dsg 1660/update profile

### PR DESCRIPTION
**Pull request by lanamckay**
# Basics

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [x] Styling and layout changes
- [ ] Code style changes (formatting, local variables)
- [ ] Refactoring (no functional changes)
- [ ] Build or CI related changes
- [ ] Tests
- [ ] Other... Please describe:

## What changes does this PR introduce?
Incorporates Form Errors component into profile updates

## Screenshot of the affected changes

Before:
Screenshot 2023-10-02 at 2 06 12 PM
After:
Screenshot 2023-10-02 at 2 06 29 PM
Screenshot 2023-10-02 at 2 06 37 PM

## Checklist

- [ ] I have updated the documentation accordingly
- [ ] I have confirmed all acceptance criteria defined by the story have been met
- [ ] I have confirmed that Sonar Qube findings for new code are reviewed and fixed, or will be whitelisted

### CODE REVIEW

File: libs/public/feature/profile/src/lib/components/profile/profile.component.ts [hunk 1]
```diff
@@ -51,23 +57,44 @@ import { PublicFeatureProfileService } from '../../public-feature-profile.servic
   styleUrls: ['./profile.component.scss'],
   templateUrl: './profile.component.html',
 })
-export class ProfileComponent implements OnDestroy {
+export class ProfileComponent {
```
Comment (asultan001) [inline libs/public/feature/profile/src/lib/components/profile/profile.component.ts:60]: @davidyoung77 does this approach make this harder to localize?

File: libs/shared/ui/nuverial/src/lib/components/checkbox/checkbox.component.spec.ts
Comment (asultan001) [inline libs/shared/ui/nuverial/src/lib/components/checkbox/checkbox.component.spec.ts:1]: Why comment out all of these tests?

Comment (lanamckay) [reply libs/shared/ui/nuverial/src/lib/components/checkbox/checkbox.component.spec.ts:1]: Hi @asultan001, those three tests were failing with my updated code.
Please see new commit with the updated tests, thanks!

File: libs/public/feature/profile/src/lib/components/profile/profile.component.ts (hunk 1 above)
Comment (davidyoung77) [reply libs/public/feature/profile/src/lib/components/profile/profile.component.ts:60]: This is good for now, we will strongly define the localization patterns in the work I am doing this sprint

File: libs/shared/ui/nuverial/src/lib/components/checkbox/checkbox.component.html [hunk 2]
```diff
@@ -1,21 +1,33 @@
...
-  [labelPosition]="labelPosition"
-  [ngClass]="{'nuverial-checkbox-invalid': invalid$ | async, 'nuverial-checkbox-forced-invalid': invalid}"
-  [required]="required"
-  (change)="onChange($event)"
->
-  <div class="nuverial-checkbox-label">
-    <ng-content select="[nuverialCardContentType='label']"></ng-content>
-  </div>
-</mat-checkbox>
+<div class="nuverial-checkbox-card-container" [ngClass]="{'image-position-before': imagePosition === 'before'}">
```
Comment (davidyoung77) [inline libs/shared/ui/nuverial/src/lib/components/checkbox/checkbox.component.html:1]: This is wrong, I have DM'd you

File: libs/shared/ui/nuverial/src/lib/components/checkbox/checkbox.component.spec.ts
Comment (davidyoung77) [inline libs/shared/ui/nuverial/src/lib/components/checkbox/checkbox.component.spec.ts:1]: These changes are likely related to the component file updates above and will change once that is fixed

File: libs/shared/ui/nuverial/src/lib/components/checkbox/checkbox.component.ts [hunk 3]
```diff
@@ -99,6 +105,15 @@ export class NuverialCheckboxComponent extends FormInputBaseDirective implements
    * Checkbox required indicator
    */
   @Input() public required = false;
+  /**
```
Comment (davidyoung77) [inline libs/shared/ui/nuverial/src/lib/components/checkbox/checkbox.component.ts:108]: why these additions?

**Review by davidyoung77 — CHANGES_REQUESTED** [review]
This needs more work, I have dm'd you

File: libs/shared/ui/nuverial/src/lib/components/checkbox/checkbox.component.ts (hunk 3 above)
Comment (lanamckay) [reply libs/shared/ui/nuverial/src/lib/components/checkbox/checkbox.component.ts]: These changes were related to the component file.
Please see updated commit

File: libs/shared/ui/nuverial/src/lib/components/checkbox/checkbox.component.spec.ts
Comment (lanamckay) [reply libs/shared/ui/nuverial/src/lib/components/checkbox/checkbox.component.spec.ts:1]: Tests work as expected with the updated html file

File: libs/shared/ui/nuverial/src/lib/components/checkbox-card/checkbox-card-async.component.spec.ts
Comment (davidyoung77) [inline libs/shared/ui/nuverial/src/lib/components/checkbox-card/checkbox-card-async.component.spec.ts:1]: I believe these changes are un-intentional

File: libs/shared/ui/nuverial/src/lib/components/checkbox/checkbox.component.ts
Comment (davidyoung77) [inline libs/shared/ui/nuverial/src/lib/components/checkbox/checkbox.component.ts:1]: I'd like to see a unit test, testing the implementation here that we are using in the profile component (we discussed this in our meeting today)

We should be sure to check that the dom is updated properly in the test

**Review by davidyoung77 — CHANGES_REQUESTED** [review]

File: libs/shared/ui/nuverial/src/lib/components/checkbox/checkbox.component.ts
Comment (lanamckay) [reply libs/shared/ui/nuverial/src/lib/components/checkbox/checkbox.component.ts:1]: Currently in progress.
As discussed in stand up, I have created a subtask for this.

**Review by davidyoung77 — APPROVED** [review]

**Review by davidyoung77 — APPROVED** [review]

**Pull request merged**